src/httpServer/server.ts

```diff
@@ -71,19 +69,51 @@ export class OperonHttpServer {
     });
   }
 
-  static registerDecoratedEndpoints(operon : Operon, router : Router)
-  {
+  /**
+   * Register workflow recovery endpoint.
+   * Receives a list of executor IDs and returns a list of workflowUUIDs.
+   */
+  static registerRecoveryEndpoint(operon: Operon, router: Router) {
+    // Handler function that parses request for recovery.
+    const recoveryHandler = async (koaCtxt: Koa.Context, koaNext: Koa.Next) => {
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+      const executorIDs = koaCtxt.request.body as string[];
+      operon.logger.info("Recovering workflows for executors: " + executorIDs.toString());
+      const recoverHandles = await operon.recoverPendingWorkflows(executorIDs);
+      operon.recoveryWorkflowHandles.push(...recoverHandles);
+
+      // Return a list of workflowUUIDs being recovered.
+      koaCtxt.body = await Promise.allSettled(recoverHandles.map((i) => i.getWorkflowUUID())).then((results) =>
+        results.filter((i) => i.status === "fulfilled").map((i) => (i as PromiseFulfilledResult<unknown>).value)
+      );
+      await koaNext();
+    };
```

Comment:
Just to confirm: `operon.recoverPendingWorkflows()` will be called:
- At least once at startup (during `operon.init()`)
- Any time this endpoint is called

So the only thing that prevents a new Operon process to recover all `PENDING` workflows executed by the `local` executor in the cloud, is that we never expect any workflow, in the cloud, to be executed by the `local` executor.

Reply:
Yup, that's my next step. I will need to modify the cloud controller to always set an `operon-executor-id` header when dispatching the request to a microVM.